decidim-ai/lib/decidim/ai/spam_detection/strategy/bayes.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "classifier-reborn"
+
+module Decidim
+  module Ai
+    module SpamDetection
+      module Strategy
+        class Bayes < Base
+          def initialize(options = {})
+            super
+            @options = { adapter: :memory, categories: [:ham, :spam], params: {} }.deep_merge(options)
+
+            @available_categories = options[:categories]
+            @backend = ClassifierReborn::Bayes.new(*available_categories, backend: configured_backend)
+          end
+
+          def log
+            return unless category
+
+            "The Classification engine marked this as #{category}"
+          end
+
+          # Calling this method without any trained categories will throw an error
+          def untrain(category, content)
+            return unless backend.categories.collect(&:downcase).collect(&:to_sym).include?(category)
+
+            backend.untrain(category, content)
+          end
+
+          delegate :train, :reset, to: :backend
+
+          def classify(content)
```

Comment:
@ahukkanen , regarding the dataset, i know it is a small one, something just to start with. Back in 2022 i have asked the community to help me by providing some anonymized data to play with, yet only one person replied to my request. This is something that we can improve in the future. The only installation I administered at the time had a Omniauth only access, and there was not enough spam data. 

Regarding the storage system, yes, an improvement can be done. I could load the for a marshall dump when memory backend is used, and load the trained data when is needed. I have not created the module to be used with "in memory". That is being added there just for development purposes

Regarding the multiple classifier part, based on your feedback, we have 2 spam engines working in parallel ( one for resources and one for the users ). Apart of that, you have the possibility define your own analyzers, and if you really want to go beyond that, you have the possibility to register  your own resource , which can implement a new custom classifier method. 

I know this can be improved and it can evolve in the future. I aimed to provide the most simplistic module that can fight spam while allows you to customize certain elements from it.

Reply:
About the dataset size, what we've been talking is to extend the Open Data files to add the moderations. Having the moderations public is something that we discussed in the past, as it brings transparency to the admins work.

(We would still need to clean the data, as we know that in the past admins used the moderation to delete tests that they themselves have done, but that's another topic)

Reply:
Marking this as resolved, as both of the main concerns mentioned in this thread are already solved:

1. `a properly balanced (i.e. 50% spam, 50% ham) large enough dataset` -> this will be possible to solve once we have [#13529](https://github.com/decidim/decidim/issues/13529) and [#13530](https://github.com/decidim/decidim/issues/13530)
2. `we have the problem that if we are using the in-memory database (as is the default) and start adding multiple in-memory classifiers, it will lead to the application requiring a lot of memory to run` -> we have changed the database to Redis by default in all the examples and documentation